### ISSUE TITLE
Add lombok generated getters to the STOP_METHODS feature

### ISSUE THREAD

**Issue by pedrovelho** [STAMP-project/pitest-descartes#86]
#### Characteristics
- **Issue Type**: [bug]
- **Reproducibility**: [always]
- **Severity**: [feature]
- **Tool/Service/Component**: [PIT 1.4.0, Descartes 1.2.4]
- **Execution Environment**: [ubuntu 16, gradle 2.10]
- **Reporter**: [Pedro, pedro.velho@activeeon.com]

#### Description

When using lombok and gradle the automatic generated classes trigger always a mutation of the auto-generated class' constructor. This is a false positive IHMO and should be ignored.

#### Steps to reproduce

Create a simple gradle project with the following configuration.

```gradle
buildscript {
    repositories {
        mavenCentral()
        mavenLocal()
    }

    configurations.maybeCreate("pitest")

    dependencies {
        classpath 'info.solidsoft.gradle.pitest:gradle-pitest-plugin:1.3.0'
        pitest 'eu.stamp-project:descartes:1.2.4'
    }
}

plugins {
    id 'io.franzbecker.gradle-lombok' version '1.14'
    id 'java'
}

apply plugin : 'java'
apply plugin : 'application'
mainClassName = 'sample.Application'

repositories {
  mavenCentral()
}

dependencies {
  testCompile 'junit:junit:4.12'
}

apply plugin: "info.solidsoft.pitest"

pitest {
  mutationEngine = "descartes"
  pitestVersion = "1.4.0"
  targetClasses = ['sample.*']
  outputFormats = ['XML', 'HTML', 'METHODS','ISSUES']
}
```

Create a simple java program that uses lombok to auto generate classes and methods through annotations.

```java
package sample;
import lombok.Builder;
import lombok.AllArgsConstructor;

@AllArgsConstructor
@Builder
public class SomeObject {
    private String name;
}
```

Finally, create a test that will use the builder automatically created.

``` java
package sample;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

import org.junit.Test;

public class TestTestCoverage {
    @Test
	public void descartesTest () {
    		SomeObject so = SomeObject.builder().build();
		assertNotNull(so);
	}
    
}
```

The result after `gradle pitest` shows a issue as follows:

![screenshot from 2018-11-12 16-31-42](https://user-images.githubusercontent.com/9976301/48357282-7bb9ed00-e698-11e8-92a6-06533b319064.png)
```
## name(java.lang.String)
Class
SomeObject$SomeObjectBuilder
Package
sample
This method is pseudo-tested

## Transformations
The following transformations were applied but they were not detected by the test suite:

All methods instructions replaced by: return null;

## Tests
The method is covered by the following test cases:

sample.TestTestCoverage.descartesTest(sample.TestTestCoverage)
```
As we can see the constructor SomeObjectBuilder of the auto-generated builder class triggers a survived mutation.

#### [Help on issue template](https://github.com/STAMP-project/pitest-descartes/blob/master/.github/ISSUE_DOC.md) 
Preview to follow the link or open file .github/ISSUE_DOC.md

**Comment by oscarlvp** [comment]
It is not a false positive in the sense that the method `name` is indeed covered and when the transformation is done, it is not detected by the test suite.
So, since these methods are a parallel to a regular simple getter it could be interesting to detect them and avoid mutating them.
Their pattern seems to be the following:
```
 0: aload_0
 1: aload_1
 2: putfield  <field>
 5: aload_0
 6: areturn
```

So, this can be added to the `STOP_METHOD` feature exposed by Descartes.

**Comment by oscarlvp** [comment]
Solved in e7953cd399fc346f5a227d3cf9f9241cf2069221
